pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotTableIdealStateBuilder.java

```diff
@@ -209,10 +209,12 @@ private static String getGroupIdFromRealtimeDataTable(String realtimeTableName,
   private static class KafkaPartitionsCountFetcher implements Callable<Boolean> {
     private int _partitionCount = -1;
     private final StreamMetadata _streamMetadata;
+    private StreamConsumerFactory _streamConsumerFactory;
     private Exception _exception;
 
     private KafkaPartitionsCountFetcher(StreamMetadata streamMetadata) {
       _streamMetadata = streamMetadata;
+      _streamConsumerFactory = StreamConsumerFactoryProvider.create(_streamMetadata);
```

Comment:
this class should be moved to a kafka specific package. pls add a TODO

Reply:
Added TODO for this and also for KafkaOffsetFetcher in PinotLLCRealtimeSegmentManager